python/cog/server/worker.py

```diff
@@ -456,7 +480,7 @@ def _loop(
         predict: Callable[..., Any],
         redirector: StreamRedirector,
     ) -> None:
-        with scope(self._loop_scope()), redirector:
+        with scope(self._loop_scope()):
```

Comment:
I think we can lift `_loop_scope` up to where we now set up the redirector. It's not strictly identical - we now set loop scope in setup, which we didn't before - but I don't think that's a real problem.